### PR TITLE
MOD-6768: Fix FT.EXPLAIN - it returns additional `}`

### PR DESCRIPTION
**Pull request by nafraf**
**Description**
Patch to fix FT.EXPLAIN, which returns an additional `}`  in case of expressions containing:
- wildcard
- empty query
- geoshape

<details>

<summary>Example of the queries with error</summary>

```sh
127.0.0.1:6379> ft.create idx schema tag1 TAG

127.0.0.1:6379> ft.explain idx "@tag1:{w'*'}" DIALECT 2
"TAG:@tag1 {\n  WILDCARD{*}\n}\n}\n"

127.0.0.1:6379> ft.explaincli idx "@tag1:{w'*'}" DIALECT 2
1) TAG:@tag1 {
2)   WILDCARD{*}
3) }
4) }

127.0.0.1:6379> FT.explaincli idx "*" DIALECT 2
1) <WILDCARD>}

127.0.0.1:6379> FT.EXPLAIN idx "" DIALECT 2
"<empty>}\n"

127.0.0.1:6379> FT.EXPLAINcli idx '@g:[WITHIN $poly]' PARAMS 2 poly 'POLYGON((0 0, 0 1, 1 1, 0 0))' DIALECT 3
1) GEOSHAPE{1 POLYGON((0 0, 0 1, 1 1, 0 0))}
2) }
```

</details>

**Which issues this PR fixes**
1. MOD-6768

**Main objects this PR modified**
1. ...

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: src/query.c [hunk 1]
```diff
@@ -1825,24 +1840,33 @@ static sds QueryNode_DumpSds(sds s, const IndexSpec *spec, const QueryNode *qs,
...
     case QN_WILDCARD_QUERY:
       s = sdscatprintf(s, "WILDCARD{%s}\n", qs->verb.tok.str);
+      if(!printAttributes) {
+        return s;
+      }
       break;
     case QN_NULL:
       s = sdscat(s, "<empty>");
-      break;
+      return s;
```
Comment (raz-mon) [inline src/query.c:1855]: Is this a separate bug?

**Review by raz-mon — COMMENTED** [review]
Nice!
See some comments

File: src/query.c [hunk 2]
```diff
@@ -1825,24 +1840,33 @@ static sds QueryNode_DumpSds(sds s, const IndexSpec *spec, const QueryNode *qs,
       break;
     case QN_WILDCARD:
       s = sdscat(s, "<WILDCARD>");
-      break;
+      return s;
```
[the remark this answers is not in the document]
Comment (nafraf) [reply src/query.c]: It's the same bug:
```
127.0.0.1:6379> FT.explaincli idx "*" DIALECT 2
1) <WILDCARD>}
```

File: src/query.c [hunk 3]
```diff
@@ -1685,6 +1685,8 @@ static sds QueryNode_DumpSds(sds s, const IndexSpec *spec, const QueryNode *qs,
     s = sdscat(s, "@NULL:");
   }
 
+  bool curlyBracketIsOpen = false;
```
[the remark this answers is not in the document]
Comment (nafraf) [reply src/query.c]: I removed that flag, Omer suggested to move the closing of the curly brackets near to its opening.

File: tests/pytests/test.py [hunk 4]
```diff
@@ -521,8 +521,9 @@ def testExplain(env):
 
     env.expect(
         'ft.create', 'idx', 'ON', 'HASH',
-        'schema', 'foo', 'text', 'bar', 'numeric', 'sortable',
-        'v', 'VECTOR', 'HNSW', '6', 'TYPE', 'FLOAT32', 'DIM', '2','DISTANCE_METRIC', 'L2', 't', 'TEXT').ok()
+        'schema', 't', 'TEXT', 'bar', 'numeric', 'sortable',
```
[the remark this answers is not in the document]
Comment (nafraf) [reply tests/pytests/test.py]: OK. Done. Thanks.

**Review by raz-mon — APPROVED** [review]
👌

**Pull request merged**